src/apps/api/views/competitions.py

```diff
@@ -347,9 +368,27 @@ def perform_create(self, serializer):
         self._ensure_organizer_participants_accepted(instance)
 
     def perform_update(self, serializer):
+        instance = self.get_object()
+
+        initial_tasks = {}
+        for phase in instance.phases.all():
+            initial_tasks.update({phase.id: set(phase.tasks.all())})
```

Comment:
This could be improved by using a `prefetch_related` and dictionary comprehension.
```suggestion
        initial_tasks = {phase.id: set(phase.tasks.all()) for phase in instance.phases.all().prefetch_related('tasks')}
```